### PR TITLE
UI Tests chat box, Text box and Button

### PR DESCRIPTION
**Pull request by manpreetkaurnventive**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?

UI Tests related to chat box, Text box and Button.

## What is the current behavior?
UI tests doen not exists.

## What is the new behavior?
Added new UI tests.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [X] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):
https://nventive.visualstudio.com/Umbrella/_sprints/taskboard/Umbrella%20Team/Umbrella/2020/Sprint%202?workitem=172736
https://nventive.visualstudio.com/Umbrella/_sprints/taskboard/Umbrella%20Team/Umbrella/2020/Sprint%202?workitem=172616
https://nventive.visualstudio.com/Umbrella/_sprints/taskboard/Umbrella%20Team/Umbrella/2020/Sprint%202?workitem=173200

### CODE REVIEW

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 1]
```diff
@@ -256,5 +256,91 @@ void Check(bool expected, string msg)
...
+
+			// Again click on toggleSwitch02 and verify text and number of clicks in all text blocks
+			toggleSwitch02.Tap();
+			verifyTextBlockText("ToggleSwitch toggleSwitch02 Toggled True (1)", "No value", "No command");
+
+			// Again click on toggleSwitch02 and verify text and number of clicks in all text blocks
+			toggleSwitch02.Tap();
+			verifyTextBlockText("ToggleSwitch toggleSwitch02 Toggled False (2)", "No value", "No command");
+		}
+		private void verifyTextBlockText(string text1, string text2, string text3)
```
Comment (agneszitte-nventive) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]: Space on top ;)

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/TextBox/TextBox_Disabled_State.xaml [hunk 2]
```diff
@@ -18,11 +18,13 @@
 				<StackPanel>
 					<ToggleSwitch IsOn="{Binding [IsEnabled], Mode=TwoWay}"
                                   OnContent="TextBox is Enabled"
```
Comment (agneszitte-nventive) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/TextBox/TextBox_Disabled_State.xaml]: Align properties with IsOn (Ctrl + K & Ctrl + D, in order to format the document) ;)

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/TextBox/TextBox_Disabled.xaml [hunk 3]
```diff
@@ -11,12 +11,12 @@
 
 	<StackPanel Margin="50">
 		<TextBox PlaceholderText="disables on letter f..."
-					 x:Name="DisableOnf"
+					 x:Name="DisableOnfTextBox"
```
Comment (agneszitte-nventive) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/TextBox/TextBox_Disabled.xaml]: Align properties with PlaceholderText (Ctrl + K & Ctrl + D, in order to format the document) ;)

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 4]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+			// Again click on toggleSwitch02 and verify text and number of clicks in all text blocks
+			toggleSwitch02.Tap();
+			verifyTextBlockText("ToggleSwitch toggleSwitch02 Toggled True (1)", "No value", "No command");
+
+			// Again click on toggleSwitch02 and verify text and number of clicks in all text blocks
+			toggleSwitch02.Tap();
+			verifyTextBlockText("ToggleSwitch toggleSwitch02 Toggled False (2)", "No value", "No command");
+		}
+
+		private void verifyTextBlockText(string text1, string text2, string text3)
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
		private void VerifyTextBlockText(string text1, string text2, string text3)
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 5]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+		}
+
+		[Test]
+		[AutoRetry]
+		public void ButtonNative_ToggleSwitch_Valiation()
+		{
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.Buttons_Native");
+
+			// Verify text of all text blocks initially
+			verifyTextBlockText("No value", "No value", "No command");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("No value", "No value", "No command");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 6]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+
+			var toggleSwitch02 = _app.Marked("toggleSwitch02");
+			var enableToggleSwitch02 = _app.Marked("enableToggleSwitch02");
+
+			// Click on toggleSwitch02 and verify text- It should not change as toggleSwitch02 is not enabled
+			toggleSwitch02.Tap();
+			verifyTextBlockText("No value", "No value", "No command");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("No value", "No value", "No command");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 7]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+
+			enableToggleSwitch02.Tap();
+			verifyTextBlockText("No value", "No value", "No command");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("No value", "No value", "No command");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 8]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+
+			// Again click on toggleSwitch02 and verify text and number of clicks in all text blocks
+			toggleSwitch02.Tap();
+			verifyTextBlockText("ToggleSwitch toggleSwitch02 Toggled True (1)", "No value", "No command");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("ToggleSwitch toggleSwitch02 Toggled True (1)", "No value", "No command");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 9]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+
+			// Again click on toggleSwitch02 and verify text and number of clicks in all text blocks
+			toggleSwitch02.Tap();
+			verifyTextBlockText("ToggleSwitch toggleSwitch02 Toggled False (2)", "No value", "No command");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("ToggleSwitch toggleSwitch02 Toggled False (2)", "No value", "No command");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 10]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.Buttons_Native");
+
+			// Verify text of all text blocks initially
+			verifyTextBlockText("No value", "No value", "No command");
+
+			var button01 = _app.Marked("button01");
+
+			// Click on button01 and verify text and number of clicks in all text blocks
+			button01.Tap();
+			verifyTextBlockText("Button button01 Clicked (1)", "Button button01 Tapped (1)", "Command Button 01 (1)");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("Button button01 Clicked (1)", "Button button01 Tapped (1)", "Command Button 01 (1)");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 11]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+
+			// Again click on button01 and verify text and number of clicks in all text blocks
+			button01.Tap();
+			verifyTextBlockText("Button button01 Clicked (2)", "Button button01 Tapped (2)", "Command Button 01 (2)");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("Button button01 Clicked (2)", "Button button01 Tapped (2)", "Command Button 01 (2)");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 12]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+
+			// Verify text of all text blocks initially
+			verifyTextBlockText("No value", "No value", "No command");
+
+			var button02 = _app.Marked("button02");
+			var enableButton02 = _app.Marked("enableButton02");
+
+			// Click on button02 and verify text- It should not change as button 02 is not enabled
+			button02.Tap();
+			verifyTextBlockText("No value", "No value", "No command");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("No value", "No value", "No command");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 13]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+		}
+
+		[Test]
+		[AutoRetry]
+		public void ButtonNative_Button02_Valiation()
+		{
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.Buttons_Native");
+
+			// Verify text of all text blocks initially
+			verifyTextBlockText("No value", "No value", "No command");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("No value", "No value", "No command");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 14]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+
+			var button02 = _app.Marked("button02");
+			var enableButton02 = _app.Marked("enableButton02");
+
+			// Click on button02 and verify text- It should not change as button 02 is not enabled
+			button02.Tap();
+			verifyTextBlockText("No value", "No value", "No command");
+
+			enableButton02.Tap();
+			verifyTextBlockText("No value", "No value", "No command");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("No value", "No value", "No command");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 15]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+
+			// Again click on button02 and verify text and number of clicks in all text blocks
+			button02.Tap();
+			verifyTextBlockText("Button button02 Clicked (1)", "Button button02 Tapped (1)", "Command Button 02 (1)");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("Button button02 Clicked (1)", "Button button02 Tapped (1)", "Command Button 02 (1)");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 16]
```diff
@@ -257,6 +257,93 @@ void Check(bool expected, string msg)
...
+
+			// Again click on button02 and verify text and number of clicks in all text blocks
+			button02.Tap();
+			verifyTextBlockText("Button button02 Clicked (2)", "Button button02 Tapped (2)", "Command Button 02 (2)");
```
Comment (dr1rrb) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]:
```suggestion
			VerifyTextBlockText("Button button02 Clicked (2)", "Button button02 Tapped (2)", "Command Button 02 (2)");
```

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs [hunk 17]
```diff
@@ -414,5 +414,59 @@ public void TextBox_TextProperty_Validation()
...
+			var enabledCheckBox = _app.Marked("EnabledCheckBox");
+
+			// Tap in text box and verify the disability with Checkbox
+			disabledWithCheckboxTextBox.Tap();
+			disabledWithCheckboxTextBox.EnterText("Text box is not disabled");
+			Assert.AreEqual("Text box is not disabled", disabledWithCheckboxTextBox.GetDependencyPropertyValue("Text")?.ToString());
+			enabledCheckBox.Tap();
+
+			disabledWithCheckboxTextBox.ClearText();
+			Assert.AreNotEqual(" ", disabledWithCheckboxTextBox.GetDependencyPropertyValue("Text")?.ToString());
```
Comment (XiaotianNetlift) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs]: A cleared textbox should not be a space `" "`. use `string.Empty` or `""`.

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs (hunk 4 above)
Comment (agneszitte-nventive) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]: Change done, thanks @dr1rrb for noticing ;) !

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs [hunk 18]
```diff
@@ -414,5 +414,59 @@ public void TextBox_TextProperty_Validation()
...
+
+			var toggleSwitch = _app.Marked("ToggleSwitch");
+			var toggleControlledTextBox = _app.Marked("ToggleControlledTextBox");
+
+			// Tap in text box and verify the disability with toggle button
+			toggleControlledTextBox.Tap();
+			toggleControlledTextBox.EnterText("Text box is not disabled");
+			Assert.AreEqual("Text box is not disabled", toggleControlledTextBox.GetDependencyPropertyValue("Text")?.ToString());
+			toggleSwitch.Tap();
+			Assert.AreNotEqual(" ", toggleControlledTextBox.GetDependencyPropertyValue("Text")?.ToString());
```
Comment (XiaotianNetlift) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs]: Same as above. And, are we not missing a "clear" after the "tap"?

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs (hunk 17 above)
Comment (agneszitte-nventive) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs]: Change done ;) thanks for noticing @XiaotianNetlift !

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs (hunk 18 above)
Comment (agneszitte-nventive) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs]: Change done and yes was missing a cleartext ;) thanks for noticing @XiaotianNetlift !

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs (hunk 17 above)
Comment (dr1rrb) [reply src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/TextBoxTests/TextBoxTests.cs]: `Assert.IsTrue(string.IsEmptyOrWhitespace(../..))` 😎

File: src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs [hunk 19]
```diff
@@ -257,6 +257,96 @@ void Check(bool expected, string msg)
...
+		{
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.Buttons_Native");
+
+			// Verify text of all text blocks initially
+			VerifyTextBlockText("No value", "No value", "No command");
+
+			var button01 = _app.Marked("button01");
+
+			// Click on button01 and verify text and number of clicks in all text blocks
+			button01.Tap();
```
Comment (carldebilly) [inline src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/ButtonTests/UnoSamples_Tests.Button_IsEnabled_Automated.cs]: `.FastTap()` instead.
```suggestion
			button01.FastTap();
```
> \+ other places...

**Comment by jeromelaban** [comment]
These behaviors are validated by more recent UI tests

**Pull request closed without merging**